packages/openapi-generator/src/generator.ts

```diff
@@ -141,3 +110,96 @@ async function generateOpenApiService(
 function parseServiceName(filePath: string): string {
   return parse(filePath).name.replace(/-openapi$/, '');
 }
+
+/**
+ * Checks if a service already exists in the output directory.
+ * @param serviceName The name of the service to be searched.
+ * @param outputDir Path to the output directory.
+ * @returns 'true' if a duplicate service is already in the output directory.
+ */
+async function duplicateServiceExists(
+  serviceName: string,
+  outputDir: string
+): Promise<boolean> {
+  const currentOutput: string[] = await readdir(outputDir);
+  return currentOutput.some(fileName => {
+    if (serviceName === fileName) {
+      return true;
+    }
+  });
+}
+
+/**
+ * Generates an OpenAPI Service from a file.
+ * @param filePath The filepath where the service to generate is located.
+ * @param options  Options to configure generation.
+ */
+async function generateFromFile(
+  filePath: string,
+  options: GeneratorOptions
+): Promise<void> {
+  const serviceName = parseServiceName(filePath);
+
+  let serviceDir: string;
+  if (await duplicateServiceExists(serviceName, options.outputDir)) {
+    serviceDir = resolve(
+      options.outputDir,
+      basename(dirname(filePath)) + '-' + serviceName
+    );
+  } else {
+    serviceDir = resolve(options.outputDir, serviceName);
+  }
+
+  let openApiDocument;
+  try {
+    openApiDocument = await convertOpenApiSpec(filePath);
+  } catch (err) {
+    logger.error(
+      `Could not convert document at ${filePath} to the format needed for parsing and generation. Skipping service generation.`
+    );
+    return;
+  }
+  const convertedInputFilePath = resolve(serviceDir, 'open-api.json');
+  const parsedOpenApiDocument = await parseOpenApiDocument(
+    openApiDocument,
+    serviceName
+  );
+
+  if (!parsedOpenApiDocument.operations.length) {
+    logger.warn(
+      `The given OpenApi specificaton does not contain any operations. Skipping generation for input file: ${filePath}`
+    );
+    return;
+  }
+
+  await mkdir(serviceDir, { recursive: true });
+  await writeFile(
+    convertedInputFilePath,
+    JSON.stringify(openApiDocument, null, 2)
+  );
+
+  await generateOpenApiService(convertedInputFilePath, serviceDir);
+  await generateSDKSources(serviceDir, parsedOpenApiDocument, options);
+}
+
+/**
+ * Recursively searches through a given input path and returns all file paths as a string array.
+ * @param input the path to the input directory.
+ * @returns all file paths as a string array.
+ */
+async function recursiveInputPathSearch(input: string): Promise<string[]> {
```

Comment:
You will also find this in the styleguide soon 👍
Functions should start with a verb. I think the fact that it is recursive does not need to be a part of the function name.
Consider `getInputFilePaths` or something in that direction.

Reply:
I see, I changed it